GLTFExporter Fix Morph

takahirox opened the pull request:
This PR fixes `GLTFExporter` Morph.

- glTF 2.0 morph supports only POSITION/NORMAL/TANGENT. So let's skip others.
- Three.js morph attribute has absolute values while the one of glTF has relative values. So we need to convert.

You can see that the second one fixes a morph issue on Don's glTF viewer
https://gltf-viewer.donmccurdy.com/

Without this change [scene.zip](https://github.com/mrdoob/three.js/files/1585993/scene.zip)
With this change [scene_fix.zip](https://github.com/mrdoob/three.js/files/1585994/scene_fix.zip)

Zip files include glTF AnimationMorphCube model loaded with `GLTFLoader` and then exported with `GLTFExporter`

https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/AnimatedMorphCube

/cc @fernandojsg @donmccurdy

takahirox committed:
GLTFExporter Fix Morph

donmccurdy commented on examples/js/exporters/GLTFExporter.js:
```diff
@@ -800,9 +800,41 @@ THREE.GLTFExporter.prototype = {
 
+						// glTF 2.0 morph supports only POSITION/NORMAL/TANGENT.
+						//
+						// @TODO TANGENT support
```
TBH I wouldn't bother adding a @\TODO about this since three.js doesn't support stored tangents currently...

donmccurdy commented on examples/js/exporters/GLTFExporter.js:
```diff
@@ -800,9 +800,41 @@ THREE.GLTFExporter.prototype = {
+
+						var baseAttribute = geometry.attributes[ attributeName ];
+						// Clones attribute not to override
+						var attribute2 = attribute.clone();
```
nit: could you rename this `relativeAttribute` or something like that? Comment above it wouldn't be necessary in that case.

donmccurdy reviewed (approved):
Nice, thanks!

takahirox committed:
Apply the review comment to GLTFExporter

takahirox commented:
Updated. Thanks for the comments.

Regarding absolute/relative comment, let me keep it because I think it'll be helpful for new developers.
And I've added warning message for non-POSITION/NORMAL morph.

takahirox commented:
/ping @fernandojsg

mrdoob commented:
Sorry, I lost track of this PR. I'll merge for now, we can amend if @fernandojsg has any comments.

mrdoob commented:
Thanks!
